securedrop/journalist_app/forms.py

```diff
@@ -2,8 +2,34 @@
 
 from flask_babel import gettext
 from flask_wtf import FlaskForm
-from wtforms import TextAreaField
-from wtforms.validators import InputRequired
+from wtforms import (TextAreaField, TextField, BooleanField, HiddenField,
+                     ValidationError)
+from wtforms.validators import InputRequired, Optional, Length
+
+from db import Journalist
+
+
+def otp_secret_validation(form, field):
+    strip_whitespace = field.data.replace(' ', '')
+    if len(strip_whitespace) != 40:
+        raise ValidationError(gettext('Field must be 40 characters long.'))
+
+
+class NewUserForm(FlaskForm):
+    username = TextField('username', validators=[
+        InputRequired(message=gettext('This field is required.')),
+        Length(min=Journalist.MIN_USERNAME_LEN,
+               message=gettext('Field must be at least {min_chars} '
+                               'characters long.'.format(
+                                   min_chars=Journalist.MIN_USERNAME_LEN))),
```

Comment:
Same as above (i.e. hint about the number of characters according to the program).